### PR TITLE
Datagrid toolbar

### PR DESCRIPTION
**Pull request by cchaos**
@constancecchen 

Here's a quick PR (I've detailed the changes in the commits). Mostly just some design tweaks, adding tooltips, and updating the examples to showcase "Real world" examples.

Let me know if you have any questions.

### CODE REVIEW

File: src/components/datagrid/controls/_data_grid_display.scss [hunk 1]
```diff
@@ -1,3 +1,3 @@
-.euiDataGrid__displayFormRow {
-  width: $euiFormMaxWidth;
+.euiDataGrid__stylePopoverPanel {
```
Comment (constancecchen) [inline src/components/datagrid/controls/_data_grid_display.scss:1]: Can we prefer not using the `style` naming for future-proof-ness? Once we add the row height switcher, this panel is no longer affecting just `gridStyles`, so I think a more generic `display` name makes sense

```suggestion
.euiDataGrid__displayPopoverPanel {
```

File: src/components/datagrid/controls/style_selector.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ export const useDataGridStyleSelector = (
       closePopover={() => setIsOpen(false)}
       anchorPosition="downRight"
       panelPaddingSize="s"
-      panelClassName="euiDataGrid__controlPopover"
+      panelClassName="euiDataGrid__stylePopoverPanel"
```
[the remark this answers is not in the document]
Comment (cchaos) [reply src/components/datagrid/controls/style_selector.tsx:80]: That transform stuff is actually only needed on popovers whose contents contains the drag-and-drop functionality like the sort and column selector. This new popover is just a simple form.

[the remark this answers is not in the document]
Comment (constancecchen) [reply src/components/datagrid/controls/style_selector.tsx]: Ohhh that's super good to know! Now I feel like I should have named that CSS class better. I'll add it in after this merges

File: src/components/datagrid/controls/data_grid_toolbar.tsx [hunk 3]
```diff
@@ -48,11 +48,18 @@ export const EuiDataGridToolbar = ({
...
+    if (isFullScreen && typeof document !== 'undefined') {
+      document.body.classList.add(GRID_IS_FULLSCREEN_CLASSNAME);
+
+      return () => {
+        if (typeof document !== 'undefined') {
+          document.body.classList.remove(GRID_IS_FULLSCREEN_CLASSNAME);
+        }
+      };
+    }
+  }, [isFullScreen]);
```
Comment (constancecchen) [inline src/components/datagrid/controls/data_grid_toolbar.tsx:63]: Super awesome fix, thanks! Do you think this needs a changelog entry or probably not?

File: src/components/datagrid/controls/style_selector.tsx [hunk 4]
```diff
@@ -64,6 +65,11 @@ export const useDataGridStyleSelector = (
     ...userGridStyles,
   };
 
+  const buttonLabel = useEuiI18n(
+    'euiStyleSelector.buttonText',
+    'Display options'
+  );
```
Comment (constancecchen) [inline src/components/datagrid/controls/style_selector.tsx:71]: 💯 Nice, thanks for the copy-fu!

File: src/components/datagrid/controls/_data_grid_toolbar.scss [hunk 5]
```diff
@@ -28,18 +31,9 @@
   }
 }
 
-.euiDataGrid__controlBtn {
```
Comment (constancecchen) [inline src/components/datagrid/controls/_data_grid_toolbar.scss:31]: Fantastic, thanks for cleaning this up!

File: src/components/datagrid/controls/data_grid_toolbar.tsx (hunk 3 above)
Comment (cchaos) [reply src/components/datagrid/controls/data_grid_toolbar.tsx:63]: Wouldn't hurt to give it a callout

Comment (cchaos) [reply src/components/datagrid/controls/data_grid_toolbar.tsx:63]: And I think that actually closes this: https://github.com/elastic/eui/issues/5092

Comment (constancecchen) [reply src/components/datagrid/controls/data_grid_toolbar.tsx:63]: Nice, I added a changelog entry and a closes commit (which I think should auto close on merge to main?)

Comment (cchaos) [reply src/components/datagrid/controls/data_grid_toolbar.tsx:63]: I honestly don't know if a commit message will. But might as well add it to your PR summary as well.

**Pull request merged**